FreeRTOS/FreeRTOS-Kernel
# Fix RP2040 compile warning

Fix Pico compile warning

Description
-----------

Fixed more warnings mentioned by #731.

Brief:
```sh
In file included from ../3rdParty/FreeRTOS-Kernel/include/FreeRTOS.h:64,
                 from ../3rdParty/FreeRTOS-Kernel/portable/ThirdParty/GCC/RP2040/port.c:34:
../3rdParty/FreeRTOS-Kernel/portable/ThirdParty/GCC/RP2040/port.c: In function 'vYieldCore':
../3rdParty/FreeRTOS-Kernel/portable/ThirdParty/GCC/RP2040/port.c:533:26: warning: comparison of integer expressions of different signedness: 'int' and 'uint' {aka 'unsigned int'} [-Wsign-compare]
  533 |     configASSERT(xCoreID != portGET_CORE_ID());
                                                   ^
[113/114] Building C object CMakeFiles/JadeDrive.dir/3rdParty/FreeRTOS-Kernel/tasks.c.obj
../3rdParty/FreeRTOS-Kernel/tasks.c: In function 'vTaskGetInfo':
../3rdParty/FreeRTOS-Kernel/tasks.c:5549:40: warning: assignment discards 'volatile' qualifier from pointer target type [-Wdiscarded-qualifiers]
 5549 |             pxTaskStatus->pxTopOfStack = pxTCB->pxTopOfStack;
```

Test Steps
-----------

Add below compile options and pass the compilation.
```
target_compile_options( ${TARGET} PUBLIC
    ### Gnu/Clang C Options
    $<$<COMPILE_LANG_AND_ID:C,GNU>:-fdiagnostics-color=always>
    $<$<COMPILE_LANG_AND_ID:C,Clang>:-fcolor-diagnostics>

    $<$<COMPILE_LANG_AND_ID:C,Clang,GNU>:-Wall>
    $<$<COMPILE_LANG_AND_ID:C,Clang,GNU>:-Wextra>
    $<$<COMPILE_LANG_AND_ID:C,Clang,GNU>:-Werror>
    $<$<COMPILE_LANG_AND_ID:C,Clang>:-Weverything>
)
```

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
